src/recorder.js

```diff
@@ -0,0 +1,53 @@
+var remix = require('ethereum-remix')
+var EventManager = remix.lib.EventManager
+
+class Recorder {
+  constructor (opts = {}) {
+    var self = this
+    self._api = opts.api
+    self.event = new EventManager()
+    self.data = { journal: [], _pending: {} }
+    opts.events.executioncontext.register('contextChanged', function () {
+      self.clearAll()
+    })
+    var counter = 1
```

Comment:
Why not starting from 0? cause the first account will <account - 2>... see line 18